jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/MetaDataBuilder.java

```diff
@@ -162,14 +163,6 @@ else if (value != null)
                     break;
 
                 case HOST:
-                    // :authority fields must come first.  If we have one, ignore the host header as far as authority goes.
-                    if (_authority == null)
-                    {
-                        if (field instanceof HostPortHttpField)
-                            _authority = (HostPortHttpField)field;
-                        else if (value != null)
-                            _authority = new AuthorityHttpField(value);
-                    }
```

Comment:
I don't believe it is necessary anymore because the `Request.setMetaData` method is able to take the authority from either the URI or the host field.
There is no need to look for a host header in a HTTP2 request when the authority field is mandatory.

We had this kind of dual approach were we were trying to make the h2 metadata look a bit like h1 meta data, but not entirely.... so the request still has to be aware of the differences.  So either we make metadata identical regardless of protocol - OR we just let it record the info that was in the message and let the client/server deal with it.   I've opted for the later.

Reply:
I have checked the proxy tests for h2->http/1.1 and the host header is correctly added by the HttpClient.